capplets/appearance/appearance-main.c

```diff
@@ -204,9 +206,15 @@ main (int argc, char **argv)
   if (wallpaper_files && !install_filename && !start_page)
     start_page = g_strdup ("background");
 
-  GtkNotebook* nb = GTK_NOTEBOOK(appearance_capplet_get_widget (data, "main_notebook"));
+  nb = GTK_NOTEBOOK (appearance_capplet_get_widget (data, "main_notebook"));
   gtk_widget_add_events (GTK_WIDGET (nb), GDK_SCROLL_MASK);
-  g_signal_connect (GTK_WIDGET (nb), "scroll-event",
+  g_signal_connect (nb, "scroll-event",
+                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
+                    GTK_WINDOW (w));
+
+  nb_custom_theme = GTK_NOTEBOOK(appearance_capplet_get_widget (data, "notebook2"));
+  gtk_widget_add_events (GTK_WIDGET (nb_custom_theme), GDK_SCROLL_MASK);
+  g_signal_connect (nb_custom_theme, "scroll-event",
                     G_CALLBACK (capplet_dialog_page_scroll_event_cb),
                     GTK_WINDOW (w));
```

Comment:
It should be called after:
```
g_option_context_free (option_context);
```
Don't cast GtkWidget* to GtkNotebook*:
```C
capplets/appearance/appearance.h:#define appearance_capplet_get_widget(x, y) (GtkWidget*) gtk_builder_get_object(x->ui, y)
```
```C
nb_custom_theme = appearance_capplet_get_widget (data, "notebook2");
gtk_widget_add_events (nb_custom_theme, GDK_SCROLL_MASK);
```

Reply:
Same code in "appearance-main.c" than "mate-window-properties.c"

Reply:
nb is used on argument parsing, nb_custom_theme is not.

gint gtk_notebook_page_num (GtkNotebook *notebook, GtkWidget *child);
void gtk_notebook_set_current_page (GtkNotebook *notebook, gint page_num);

https://github.com/mate-desktop/mate-control-center/blob/3c7d68bdee43165d963a340f58ec4ea8c4ac7449/capplets/appearance/appearance-main.c#L221-L236

Reply:
nb you speak is local in function 'main_window_response' and stay untouch.

Reply:
updated just now

Reply:
compiler warnings:
```
$ ./autogen.sh --enable-compile-warnings=maximum --prefix=/usr && make
<cut>
appearance-main.c: In function ‘main’:
appearance-main.c:231:39: warning: passing argument 1 of ‘gtk_notebook_page_num’ from incompatible pointer type [-Wincompatible-pointer-types]
  231 |       pindex = gtk_notebook_page_num (nb, w);
      |                                       ^~
      |                                       |
      |                                       GtkWidget * {aka struct _GtkWidget *}
In file included from /usr/include/gtk-3.0/gtk/gtk.h:155,
                 from appearance.h:24,
                 from appearance-main.c:23:
/usr/include/gtk-3.0/gtk/gtknotebook.h:186:56: note: expected ‘GtkNotebook *’ {aka ‘struct _GtkNotebook *’} but argument is of type ‘GtkWidget *’ {aka ‘struct _GtkWidget *’}
  186 | gint       gtk_notebook_page_num         (GtkNotebook *notebook,
      |                                           ~~~~~~~~~~~~~^~~~~~~~
appearance-main.c:233:40: warning: passing argument 1 of ‘gtk_notebook_set_current_page’ from incompatible pointer type [-Wincompatible-pointer-types]
  233 |         gtk_notebook_set_current_page (nb, pindex);
      |                                        ^~
      |                                        |
      |                                        GtkWidget * {aka struct _GtkWidget *}
<cut>
```

Reply:
ok, give a lot of time to test here...

Reply:
Should be correct now.